t/t1091-sparse-checkout-builtin.sh

```diff
@@ -642,4 +642,42 @@ test_expect_success MINGW 'cone mode replaces backslashes with slashes' '
 	check_files repo/deep a deeper1
 '
 
+test_expect_success 'cone mode clears ignored subdirectories' '
+	rm repo/.git/info/sparse-checkout &&
+
+	# NEEDSWORK: --sparse-index is required for now
+	git -C repo sparse-checkout init --cone --sparse-index &&
+	git -C repo sparse-checkout set deep/deeper1 &&
+
+	cat >repo/.gitignore <<-\EOF &&
+	obj/
+	*.o
+	EOF
+
+	git -C repo add .gitignore &&
+	git -C repo commit -m ".gitignore" &&
+
+	for dir in  folder1 obj folder1/obj deep/deeper2 deep/deeper2/obj
+	do
+		mkdir repo/$dir || return 1
```

Comment:
Why not `mkdir -p obj folder1/obj deep/deeper2/obj`?

Reply:
I didn't know about `-p` thanks!